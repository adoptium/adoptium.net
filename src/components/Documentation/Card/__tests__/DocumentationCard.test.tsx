import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import { FaDownload } from "react-icons/fa"
import DocumentationCard from ".."

describe("DocumentationCard component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <DocumentationCard
        title="Test Title"
        Icon={FaDownload}
        links={[
          { name: "Link1", link: "/link1" },
          { name: "Link2", link: "https://external-link" },
        ]}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it("does not render without a title", () => {
    const { container } = render(
      <DocumentationCard
        title=""
        Icon={FaDownload}
        links={[
          { name: "Link1", link: "/link1" },
          { name: "Link2", link: "https://external-link" },
        ]}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it("does not render without a Icon", () => {
    const { container } = render(
      <DocumentationCard
        title="Test Title"
        // @ts-expect-error Testing with invalid Icon prop
        Icon=""
        links={[
          { name: "Link1", link: "/link1" },
          { name: "Link2", link: "https://external-link" },
        ]}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it("does not render when links is not object", () => {
    const { container } = render(
      <DocumentationCard
        title="Test Title"
        Icon={FaDownload}
        // @ts-expect-error Testing with invalid links prop
        links="Bad Links"
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
