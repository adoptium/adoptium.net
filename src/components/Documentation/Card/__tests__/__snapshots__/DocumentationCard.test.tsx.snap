// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DocumentationCard component > does not render when links is not object 1`] = `<div />`;

exports[`DocumentationCard component > does not render without a Icon 1`] = `<div />`;

exports[`DocumentationCard component > does not render without a title 1`] = `<div />`;

exports[`DocumentationCard component > renders correctly 1`] = `
<div>
  <div
    class="w-full lg:w-1/3 p-3 group"
  >
    <div
      class="bg-gradient-to-br from-[#200E46]/90 to-[#2B1A4F]/90 rounded-3xl border border-white/20 shadow-lg overflow-hidden transition-all duration-300 hover:shadow-xl hover:shadow-[#FF1464]/10 hover:border-[#FF1464]/30 h-full transform hover:-translate-y-1"
    >
      <div
        class="relative p-6"
      >
        <div
          class="absolute -top-10 -right-10 w-24 h-24 bg-gradient-to-br from-[#FF1464]/30 to-[#FF4B54]/20 rounded-full blur-xl opacity-50"
        />
        <div
          class="relative"
        >
          <p
            class="mb-5 text-xl font-semibold flex items-center"
          >
            <span
              class="inline-flex items-center justify-center bg-[#2B1A4F] p-3 rounded-xl mr-3 shadow-inner border border-white/10"
            >
              <svg
                class="text-[#FF1464] h-5 w-5"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                />
              </svg>
            </span>
            <span
              class="text-white/90"
            >
              Test Title
            </span>
          </p>
          <div
            class="divide-y divide-white/10"
          >
            <a
              class="block px-4 py-3 text-white/80 hover:text-white hover:bg-white/10 rounded-lg my-1 transition-all duration-200"
              href="/link1"
            >
              Link1
            </a>
            <a
              class="px-4 py-3 flex justify-between items-center text-white/80 hover:text-white hover:bg-white/10 rounded-lg my-1 transition-all duration-200"
              href="https://external-link"
              rel="noopener noreferrer"
              target="_blank"
            >
              Link2
               
              <svg
                class="ml-2 text-[#FF1464]/80"
                fill="currentColor"
                height="13"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
